cmd/frieza/cli_clean.go

```diff
@@ -11,72 +11,64 @@ import (
 		WithOption(cliConfigPath()).
 		WithOption(cliDebug()).
 		WithAction(func(args []string, options map[string]string) int {
 			setupDebug(options)
-			clean(options["config"], &args[0], options["plan"] == "true", options["auto-approve"] == "true")
+			clean(options["config"], &args[0], options["plan"] == "true", options["auto-approve"] == "true", options["json"] == "true")
 			return 0
 		})
 }
 
-func clean(customConfigPath string, snapshotName *string, plan bool, autoApprove bool) {
+func clean(customConfigPath string, snapshotName *string, plan bool, autoApprove bool, jsonOutput bool) {
 	var configPath *string
+	if jsonOutput && !autoApprove {
+		cliFatalf(true, "Cannot use --json option without --auto-approve")
+	}
 	if len(customConfigPath) > 0 {
 		configPath = &customConfigPath
 	}
 	config, err := ConfigLoadWithDefault(configPath)
 	if err != nil {
-		log.Fatalf("Cannot load configuration: %s", err.Error())
+		cliFatalf(jsonOutput, "Cannot load configuration: %s", err.Error())
 	}
 	snapshot, err := SnapshotLoad(*snapshotName, config)
 	if err != nil {
-		log.Fatalf("Error load snapshot %s: %s", *snapshotName, err.Error())
+		cliFatalf(jsonOutput, "Error load snapshot %s: %s", *snapshotName, err.Error())
 	}
 
-	var providers []Provider
-	var objectsToDelete []Objects
+	destroyer := NewDestroyer()
 	objectsCount := 0
-
 	for _, data := range snapshot.Data {
 		profile, err := config.GetProfile(data.Profile)
 		if err != nil {
-			log.Fatalf("Error while getting profile %s: %s", data.Profile, err.Error())
+			cliFatalf(jsonOutput, "Error while getting profile %s: %s", data.Profile, err.Error())
 		}
 		provider, err := ProviderNew(*profile)
 		if err != nil {
-			log.Fatalf("Error intializing profile %s: %s", data.Profile, err.Error())
+			cliFatalf(jsonOutput, "Error intializing profile %s: %s", data.Profile, err.Error())
 		}
 		objects := ReadObjects(&provider)
 		diff := NewDiff()
 		diff.Build(&data.Objects, &objects)
-		count := ObjectsCount(&diff.Created)
-		objectsCount += count
-		if count > 0 {
-			fmt.Printf("Newly created object to delete in profile %s (%s):\n", profile.Name, provider.Name())
-			fmt.Printf(ObjectsPrint(&provider, &diff.Created))
-		} else {
-			fmt.Printf("No new object to delete in profile %s (%s)\n", profile.Name, provider.Name())
-		}
-		providers = append(providers, provider)
-		objectsToDelete = append(objectsToDelete, *&diff.Created)
+		objectsCount += ObjectsCount(&diff.Created)
+		destroyer.add(profile, &provider, &diff.Created)
 	}
 
-	if objectsCount == 0 {
-		fmt.Println("Nothing to delete, exiting")
+	destroyer.print(jsonOutput)
+	if plan || objectsCount == 0 {
 		return
 	}
-
-	if plan {
-		return
+	if jsonOutput {
+		disableLogs()
 	}
-
 	message := fmt.Sprintf("Do you really want to delete newly created resources?\n" +
 		"  Frieza will delete all resources shown above.")
 	if !confirmAction(&message, autoApprove) {
 		log.Fatal("Clean canceled")
```

Comment:
Maybe, we should use fmt package instead of log here. Because when we disabled logging, we won't see this message.

Reply:
Sure, this is intended: logging is disabled when --json is effective to avoid future (non-json) prints.
Also, --json imply --auto-apply which won't print this message anyway.